From 89ecd70c86b24d220f35d07c1a4e1151503cbafd Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Mon, 5 Jun 2023 13:32:26 +0200
Subject: [PATCH 305/841] refactor(stm32mp2-fdts): rework fw-config files

Remove useless defines, and move soc_fw-config (BL31 DT) and tos_fw
(BL32 = OP-TEE) load addresses and sizes to generated file
stm32mp257f-ev-ca35tdcid-fw-config.dtsi.
Remove useless st_mem_firewall alias.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 7a95f1a418dfcac348b20bc7f773fb37bdb4e80d refactor(stm32mp2-fdts): rework fw-config files]
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Change-Id: I356f245ef4e478cb6744a2648cf2f80c31f053c2
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/295448
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 fdts/stm32mp25-fw-config.dtsi                | 16 +---------
 fdts/stm32mp257f-ev-ca35tdcid-fw-config.dtsi | 31 +++++++++++++-------
 2 files changed, 22 insertions(+), 25 deletions(-)

diff --git a/fdts/stm32mp25-fw-config.dtsi b/fdts/stm32mp25-fw-config.dtsi
index d023378a0..27f20fc9e 100644
--- a/fdts/stm32mp25-fw-config.dtsi
+++ b/fdts/stm32mp25-fw-config.dtsi
@@ -7,16 +7,6 @@
 
 #include <platform_def.h>
 
-#ifndef DDR_SIZE
-#error "DDR_SIZE is not defined"
-#endif
-
-#define DDR_SEC_BASE	(STM32MP_DDR_BASE + 0x02000000)
-#define DDR_SEC_SIZE	0x02000000
-#define DDR_NS_BASE	(DDR_SEC_BASE + DDR_SEC_SIZE)
-#define DDR_NS_SIZE	(STM32MP_DDR_BASE + DDR_SIZE - DDR_NS_BASE)
-#define SOC_FW_CFG_BASE	(DDR_SEC_BASE - STM32MP_SOC_FW_CONFIG_MAX_SIZE)
-
 /dts-v1/;
 
 / {
@@ -42,19 +32,15 @@
 		};
 
 		soc_fw-config {
-			load-address = <0x0 SOC_FW_CFG_BASE>;
-			max-size = <STM32MP_SOC_FW_CONFIG_MAX_SIZE>;
 			id = <SOC_FW_CONFIG_ID>;
 		};
 
 		tos_fw {
-			load-address = <0x0 DDR_SEC_BASE>;
-			max-size = <DDR_SEC_SIZE>;
 			id = <BL32_IMAGE_ID>;
 		};
 	};
 
-	st_mem_firewall: st-mem-firewall {
+	st-mem-firewall {
 		compatible = "st,stm32mp2-mem-firewall";
 		#address-cells = <2>;
 		#size-cells = <2>;
diff --git a/fdts/stm32mp257f-ev-ca35tdcid-fw-config.dtsi b/fdts/stm32mp257f-ev-ca35tdcid-fw-config.dtsi
index d9d928208..3eedf590a 100644
--- a/fdts/stm32mp257f-ev-ca35tdcid-fw-config.dtsi
+++ b/fdts/stm32mp257f-ev-ca35tdcid-fw-config.dtsi
@@ -6,18 +6,29 @@
 /*
  * STM32MP25 tf-a firmware config
  * Project : eco
- * Generated by XLmx tool version 2.2 - 3/8/2023 2:34:26 PM
+ * Generated by XLmx tool version 2.2 - 6/1/2023 12:15:35 PM
  */
 
-&st_mem_firewall {
-	bl31_context: bl31-context@81ff0000 {
-		compatible = "shared-dma-pool";
-		reg = <0x00000000 0x81ff0000 0x00000000 0x10000>;
-		st,protreg = <RISAFPROT(RISAF_REG_ID(7), RIF_CID1_BF, RIF_CID1_BF, RIF_CID1_BF, RIF_SEC, RIF_ENC_EN, RIF_BREN_EN)>;
+/ {
+	dtb-registry {
+		soc_fw-config {
+			load-address = <0x0 0x81ff0000>;
+			max-size = <0x10000>;
+		};
+		tos_fw {
+			load-address = <0x0 0x82000000>;
+			max-size = <0x2000000>;
+		};
 	};
-	op_tee: op-tee@82000000 {
-		compatible = "shared-dma-pool";
-		reg = <0x00000000 0x82000000 0x00000000 0x2000000>;
-		st,protreg = <RISAFPROT(RISAF_REG_ID(8), RIF_CID1_BF, RIF_CID1_BF, 0, RIF_SEC, RIF_ENC_EN, RIF_BREN_EN)>;
+
+	st-mem-firewall {
+		bl31_context: bl31-context@81ff0000 {
+			reg = <0x0 0x81ff0000 0x0 0x10000>;
+			st,protreg = <RISAFPROT(RISAF_REG_ID(7), RIF_CID1_BF, RIF_CID1_BF, RIF_CID1_BF, RIF_SEC, RIF_ENC_EN, RIF_BREN_EN)>;
+		};
+		op_tee: op-tee@82000000 {
+			reg = <0x0 0x82000000 0x0 0x2000000>;
+			st,protreg = <RISAFPROT(RISAF_REG_ID(8), RIF_CID1_BF, RIF_CID1_BF, 0, RIF_SEC, RIF_ENC_EN, RIF_BREN_EN)>;
+		};
 	};
 };
-- 
2.47.2

